### PR TITLE
Added the ability to add filters

### PR DESCRIPTION
**Pull request by JJerome-NM**

### CODE REVIEW

**Review by justme1135 — CHANGES_REQUESTED** [review]

**Review by getmane — CHANGES_REQUESTED** [review]
1. We do not use the test package as a place for regular production code
2. Remove all the comments you currently have
3. Rewrite your code so that it makes sense without comments

General advice, please define some description of what you are aiming to achieve in this PR. Just from reading your code, it does not make sense at all to me.

File: jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java [hunk 1]
```diff
@@ -1,45 +1,63 @@
...
 import org.springframework.stereotype.Component;
 import org.springframework.web.socket.CloseStatus;
 import org.springframework.web.socket.TextMessage;
 import org.springframework.web.socket.WebSocketSession;
 import org.springframework.web.socket.handler.TextWebSocketHandler;
 
 import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
 import java.util.*;
```
[the remark this answers is not in the document]
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java]: Changed imports to normal

File: jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java [hunk 2]
```diff
@@ -1,45 +1,63 @@
 package com.jjerome.models;
 
+import com.jjerome.annotations.SocketComponentsScan;
 import com.jjerome.annotations.SocketController;
-import com.jjerome.context.SocketContext;
+import com.jjerome.annotations.SocketMapping;
+import com.jjerome.annotations.SocketMappingFilter;
+import com.jjerome.context.SocketControllersContext;
 import com.jjerome.dto.Response;
 import org.jetbrains.annotations.NotNull;
```
[the remark this answers is not in the document]
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/models/SocketApplication.java]: Changed Jetbrains @NotNull to lombok.NonNull

File: jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java [hunk 3]
```diff
@@ -18,11 +21,11 @@
 import java.lang.reflect.ParameterizedType;
 import java.util.*;
```
[the remark this answers is not in the document]
Comment (JJerome-NM) [reply jjsocket-main/src/main/java/com/jjerome/context/SocketControllersContext.java]: Changed imports to normal

File: jjsocket-main/src/test/java/com/jjerome/test/MainTests/ConnectFilter.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.jjerome.test.MainTests;
+
+import com.jjerome.models.SocketConnectionFilter;
+import org.jetbrains.annotations.NotNull;
```
[the remark this answers is not in the document]
Comment (JJerome-NM) [reply jjsocket-main/src/test/java/com/jjerome/test/MainTests/ConnectFilter.java]: Changed Jetbrains @NotNull to lombok.NonNull

**Review by justme1135 — APPROVED** [review]

**Pull request merged**